SwanProjects/bin/swan_kmspecs

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python
+# Copyright (c) SWAN Development Team.
+# Author: Omar.Zapata@cern.ch 2021
+
+"""
+This script allows to find the kernels for python2/3 and kernel spec paths for our kernel spec manager.
+
+The script run a subprocess inside the project environment trying to find the package ipykernel
+and run jupyter_path('kernels') to get the list of available paths for the differents kernels inside the environment as well.
+
+This is execute in the Create/Edit project handlers and the stdout is captured, getting the json information printed in the function generate_ksminfo()
+"""
+import argparse
+import json
+import os
+import pprint
+import subprocess
+import sys
+from distutils.spawn import find_executable
+from shutil import rmtree
+try:
+    from jupyter_core.paths import jupyter_path, get_home_dir
+except ImportError:
+    print("{'status':0, 'msg':'Package jupyter_core not found in the environment, kernel paths can not be found.'}")
+    sys.exit(1)
+
+
+def checkipykernel(python_interpreter):
+    """
+    Checks if ipykernel is available for the given python interpreter,
+    this function have to be executed inside the project environment.
+
+    Parameters
+    ----------
+    python_interpreter : str
+        full path to python interpreter.
+
+    Returns
+    -------
+    int
+        zero if ipykernel was found.
+    """
+    python_code = 'import ipykernel'
+    command = [python_interpreter, "-c", python_code]
+    proc = subprocess.Popen(command, stdout=subprocess.PIPE)
+    proc.wait()
+    proc.communicate()
+    return proc.returncode
+
+
+def check_native_kernel():
+    """
+    Checks if ipykernel is available for python2 and python3.
+
+    This routine is called inside the project environment,
+    It checks if python2/3 are available to check if the package ipykernel is installed,
+    if the package was found the results is saved in the dictionary with the results.
```

Comment:
It checks if python2/3 are available and, if they are, it checks if the package ipykernel is installed for them.